programmers/문자열내-p와-y의-갯수.js

```diff
@@ -0,0 +1,16 @@
+function solution(s){
+    var pNum = s.match(/p/gi);
+    var yNum = s.match(/y/gi);
+    
+    if (pNum === null && yNum === null) {
+      return true;
+    }
+    if (pNum === null || yNum === null) {
+        if (pNum === null && pNum != yNum) {
+            return false; }
+        if (yNum === null && pNum != yNum) {
+            return false; }
+    }
+    return (pNum.length === yNum.length) ? true : false;
```

Comment:
전부다 if문으로 코드를 짤 시 일치하더라도 그밑에 조건을 찾게된다
아래와 같이 if 가 아니라면 또 다시 if 를 쓰지않고 else를 사용하면 된다.

중괄호 위치를 신경 쓸 것
```suggestion
    if (pNum === null && yNum === null) return true;  // 리턴값이 한줄인경우 이렇게 수정가능이 가능하다.
    else {
        if (pNum === null && pNum != yNum) return false;
        else if (yNum === null && pNum != yNum)  return false; 
    }
    return (pNum.length === yNum.length) ? true : false;
```